cmd/root.go

```diff
@@ -226,44 +257,60 @@ func applyProtected(ctx context.Context, branches []shared.Branch, connection sh
 }
 
 func applyCommits(ctx context.Context, remote shared.Remote, branches []shared.Branch, defaultBranchName string, connection shared.Connection) ([]shared.Branch, error) {
-	results := []shared.Branch{}
+	var wg sync.WaitGroup
+	resultChan := make(chan shared.Branch, len(branches))
 
 	for _, branch := range branches {
-		if branch.Name == defaultBranchName || branch.IsDetached() {
-			branch.Commits = []string{}
-			results = append(results, branch)
-			continue
-		}
+		wg.Add(1)
+		go func(branch shared.Branch) {
+			defer wg.Done()
+
+			if branch.Name == defaultBranchName || branch.IsDetached() {
+				branch.Commits = []string{}
+				resultChan <- branch
+				return
+			}
 
-		if remoteHeadOid, err := connection.GetRemoteHeadOid(ctx, remote.Name, branch.Name); err == nil {
-			branch.RemoteHeadOid = SplitLines(remoteHeadOid)[0]
-		} else {
-			result, _ := connection.GetConfig(ctx, fmt.Sprintf("branch.%s.remote", branch.Name))
-			splitResults := SplitLines(result)
-			if len(splitResults) > 0 {
-				remoteUrl := splitResults[0]
-				if result, err := connection.GetLsRemoteHeadOid(ctx, remoteUrl, branch.Name); err == nil {
-					splitResults := strings.Fields(result)
-					if len(splitResults) > 0 {
-						branch.RemoteHeadOid = splitResults[0]
+			if remoteHeadOid, err := connection.GetRemoteHeadOid(ctx, remote.Name, branch.Name); err == nil {
+				branch.RemoteHeadOid = SplitLines(remoteHeadOid)[0]
+			} else {
+				result, _ := connection.GetConfig(ctx, fmt.Sprintf("branch.%s.remote", branch.Name))
+				splitResults := SplitLines(result)
+				if len(splitResults) > 0 {
+					remoteUrl := splitResults[0]
+					if result, err := connection.GetLsRemoteHeadOid(ctx, remoteUrl, branch.Name); err == nil {
+						splitResults := strings.Fields(result)
+						if len(splitResults) > 0 {
+							branch.RemoteHeadOid = splitResults[0]
+						}
 					}
 				}
 			}
-		}
 
-		oids, err := connection.GetLog(ctx, branch.Name)
-		if err != nil {
-			return nil, err
-		}
+			oids, err := connection.GetLog(ctx, branch.Name)
+			if err != nil {
+				return
```

Comment:
If the git process returns an error, the user should be informed of the error.

Reply:
Applied the same error propagation mechanism like the other coroutine.